### PR TITLE
Add functional test for checking factory with skipFirstLevel parameter

### PR DESCRIPTION
**Pull request by musienko-maxim**
### What does this PR do?
**Add checking the `skipFirstLevel` parameter in the factory by next scenario:**
- Add test repo to GitHub side. Create factory with source form test repo and `skipFirstLevel` parameter. 
- Consumes the factory and checks that project has been imported as well.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/11081

@SkorikSergey, @dmytro-ndp, @Ohrimenko1988

### CODE REVIEW

**Review by Ohrimenko1988 — APPROVED** [review]

**Review by SkorikSergey — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  private String gitHubUsername;
+
+  @Inject private Ide ide;
+
+  private TestFactory testFactory;
+  private String projectName;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java]: It's quite hard to understand what this method is doing without comments and indents. Could you, please, add some explanation into the code?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    Path entryPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").getPath());
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java]: Do we really need to add "default-spring-project" project into the test repository if we then adding into the factory template link to github project archive?

Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java]: Actually we need.
After adding the test repo - we have URLs like:
1) **SSH:** `git@github.com:gthub_user_name/name_repo`
2) **HTTPS:** `https://github.com/com:gthub_user_name/name_repo`
3) **ZIP:** `https://gthub_user_name/name_repo/archive/branch_name.zip`
So we use # 3 for the test

Comment (dmytro-ndp) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java]: In that case, IMHO, `testRepo.addContent(entryPath)` looks redundant.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  private TestFactory testFactory;
+  private String projectName;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    Path entryPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").getPath());
+    testRepo.addContent(entryPath);
+    projectName = testRepo.getName();
+    String subPathToGithubArchive = "/archive/master.zip";
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java]: `subPathToGithubArchive` is better to inline where it is used.

**Comment by dmytro-ndp** [comment]
`pathToGithubArchive` variable is more like a `urlToDownaloadZippedRepoFromGitHub`

**Review by dmytro-ndp — APPROVED** [review]

**Comment by musienko-maxim** [comment]
ci-build

**Pull request merged**